Furball.Engine/Engine/DevConsole/ImGuiConsole.cs

```diff
@@ -0,0 +1,130 @@
+using System.Collections.Generic;
+using System.Numerics;
+using System.Text;
+using Furball.Engine.Engine.Graphics.Drawables;
+using ImGuiNET;
+using Silk.NET.Input;
+
+namespace Furball.Engine.Engine.DevConsole {
+    public static class ImGuiConsole {
+        private static byte[]              _consoleBuffer = new byte[4096];
+
+        private static BlankDrawable _consoleInputCoverDrawable;
+
+        public static bool Visible = false;
+
+        public static unsafe void Initialize() {
+            FurballGame.InputManager.OnKeyDown += (_, key) => {
+                if (key == Key.F12 && !Visible)
+                    Visible = true;
+            };
+
+            var style = ImGui.GetStyle();
+
+            _consoleInputCoverDrawable = new BlankDrawable {
+                CoverClicks = true,
+                Clickable   = true,
+                Visible     = true,
+                CoverHovers = true,
+                Hoverable   = true,
+                Depth       = 0
+            };
+
+            FurballGame.DrawableManager.Add(_consoleInputCoverDrawable);
+
+            style.ScrollbarRounding = 0;
+            style.ChildRounding     = 0;
+            style.FrameRounding     = 0;
+            style.PopupRounding     = 0;
+            style.WindowRounding    = 0;
+
+            style.Colors[(int) ImGuiCol.WindowBg]         = new Vector4((40.0f / 255.0f), (5.0f / 255.0f),  (50.0f / 255.0f), 0.8f);
+            style.Colors[(int) ImGuiCol.FrameBg]          = new Vector4((50.0f / 255.0f), (10.0f / 255.0f), (90.0f / 255.0f), 0.8f);
+            style.Colors[(int) ImGuiCol.ChildBg]          = new Vector4((50.0f / 255.0f), (10.0f / 255.0f), (90.0f / 255.0f), 0.8f);
+            style.Colors[(int) ImGuiCol.TitleBgActive]    = new Vector4((70.0f / 255.0f), (10.0f / 255.0f), (90.0f / 255.0f), 0.8f);
+            style.Colors[(int) ImGuiCol.TitleBgCollapsed] = new Vector4((70.0f / 255.0f), (10.0f / 255.0f), (90.0f / 255.0f), 0.8f);
+
+            style.WindowPadding = new Vector2(8, 8);
+            style.ItemSpacing   = new Vector2(8, 8);
+            style.FramePadding  = new Vector2(8, 4);
+        }
+
+        public static unsafe void Draw() {
+            if (Visible) {
+                ImGui.SetNextWindowSize(new Vector2(520, 600), ImGuiCond.FirstUseEver);
+                ImGui.Begin("Console", ref Visible, ImGuiWindowFlags.NoScrollbar | ImGuiWindowFlags.NoCollapse);
+
+                _consoleInputCoverDrawable.Position     = ImGui.GetWindowPos();
+                _consoleInputCoverDrawable.OverrideSize = ImGui.GetWindowSize();
```

Comment:
This is not taking into account the window scale and it causes input with it to be broken when the window is resized to be larger

Reply:
right, i forgor